### PR TITLE
Add S3 encode tests

### PR DESCRIPTION
**Pull request by KalmanMeth**

### CODE REVIEW

File: pkg/pipeline/encode/encode_s3.go [hunk 1]
```diff
@@ -119,12 +118,11 @@ func (s *encodeS3) createObjectTimeoutLoop() {
 			log.Debugf("exiting createObjectTimeoutLoop because of signal")
 			return
 		case <-ticker.C:
-			log.Debugf("createObjectTimeoutLoop timer expire")
 			now := time.Now()
 			log.Debugf("time now = %v, expiryTime = %v", now, s.expiryTime)
-			if now.After(s.expiryTime) {
-				_ = s.writeObject()
-			}
```
Comment (ronensc) [inline pkg/pipeline/encode/encode_s3.go:127]: IIUC, this is the only place where `expiryTime` is read. If it is removed from here, maybe the variable can be removed altogether.

File: pkg/pipeline/encode/encode_s3_test.go [hunk 2]
```diff
@@ -98,19 +107,85 @@ func Test_EncodeS3(t *testing.T) {
...
+	expectedSubstring = "hour="
+	require.True(t, strings.Contains(objectName0, expectedSubstring))
+	expectedSubstring = "stream-id="
+	require.True(t, strings.Contains(objectName0, expectedSubstring))
+	expectedSubstring = "00000000"
+	require.True(t, strings.Contains(objectName0, expectedSubstring))
+	expectedSubstring = "00000001"
+	require.True(t, strings.Contains(fakeWriter.objectNames[1], expectedSubstring))
+	expectedSubstring = "00000002"
+	require.True(t, strings.Contains(fakeWriter.objectNames[2], expectedSubstring))
```
Comment (ronensc) [inline pkg/pipeline/encode/encode_s3_test.go:144]: Using `require.Contains()` is preferred over `require.True()`.
The advantage of  `require.Contains()` is that when the condition is not met, the error message is more verbose.

**Review by ronensc — APPROVED** [review]

File: pkg/pipeline/encode/encode_s3.go (hunk 1 above)
Comment (KalmanMeth) [reply pkg/pipeline/encode/encode_s3.go:127]: This is exactly for what the parameter is needed, to define the user's desired timeout period when to create a new object. So I don't think we can remove this.

File: pkg/pipeline/encode/encode_s3_test.go (hunk 2 above)
Comment (KalmanMeth) [reply pkg/pipeline/encode/encode_s3_test.go]: updated accordingly

**Pull request merged**